backend/api/petMatch.py

```diff
@@ -0,0 +1,133 @@
+from flask import Blueprint, jsonify, request
+from firebase_admin import firestore
+from connect_firebase import PawnderFirebase
+from models.match import Match
+from models.match import Status
+
+pet_match_api = Blueprint('pet_match_api', __name__)
+pawnder_firebase = PawnderFirebase()
+db = firestore.client()
+pet_db = db.collection("PET")
+
+# Send a match request to a pet
+@pet_match_api.post('/pets/<petId>/matches')
```

Comment:
If I understand your matching algorithm correctly, when pet A sends a matching request to pet B, a match entry with a status of "pending" is created in pet B's document. But what happens if pet B also sends a matching request to pet A? Is there a way to check if pet A has already sent a request, or will you create another match entry with a "pending" status in pet A's document?

Reply:
I was thinking the second case won't happen. When A match with B it will create a match with pending status in B. When we display pets for B to match we will display all pets that is not in B pending status list. Therefore, B cannot send a request to A. B can only accept or reject A. This will probably modify the get pet endpoint.